nss-day-cohort-39/tabloid-full-stack-coulees
# View All Users

### Changes
1. Created getAllUsers method in API repository and corresponding HttpGet route in controller
2. Hid users tab in header from Author user types
3. Created User Details view
4. Enabled User Deactivation
5. Added list of deactivated users to users list
6. Enable Reactivate User feature

### To Test
1. Run the SQL script 
`ALTER TABLE UserProfile
ADD IsApproved BIT DEFAULT 1 NOT NULL`
2. Login as an admin and view the users list
3. Login as an author ensure you cannot see the users tab in the header
4. Ensure you cannot navigate to the users list by the URL as an author `http://localhost:3000/users`
5. Try deactivating a user and ensure they no longer appear in the active users list, they do appear in the deactivated users list, and their posts no longer appear in the post list
6. Ensure you can see the deactivated users when you click the show deactivated button
7. Ensure you cannot login as a deactivated user

***NOTE: I turned off the 'launch browser' setting in the API bc I hate it.***
